### PR TITLE
Message generator for ShouldBeSubsetOf

### LINKED ISSUE #158: Message generator for ShouldBeSubsetOf
The error message is not very informative, it should say which items are additionally in the actual.

### PR DESCRIPTION
**Pull request by gabrielweyer**
I added a class called `ShouldBeSubsetOfMessageGenerator` to handle the formatting of the error message.

`ShouldBeSubsetOf<T>` now uses `Except` to determine which elements are missing.

Let me know if you want me to modify anything.

Fix #158

### CODE REVIEW

File: src/Shouldly/MessageGenerators/ShouldBeSubsetOfMessageGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        private static readonly Regex Validator = new Regex("ShouldBeSubsetOf", RegexOptions.Compiled);
+
+        public override bool CanProcess(ShouldlyAssertionContext context)
+        {
+            return Validator.IsMatch(context.ShouldMethod);
+        }
+
+        public override string GenerateErrorMessage(ShouldlyAssertionContext context)
+        {
+            const string format = @"{0} should be subset of {1} but {2} do not";
```
Comment (JakeGinnivan) [inline src/Shouldly/MessageGenerators/ShouldBeSubsetOfMessageGenerator.cs]: Maybe one of the following instead (better English)

 - `{0} should be subset of {1} but {2} are outside subset`
 - `{0} should be subset of {1} but {2} are not`
 - `{0} should be subset of {1} but {2} are additional`

Not quite sure which is best. @robdmoore @xerxesb any thoughts?

Comment (gabrielweyer) [inline src/Shouldly/MessageGenerators/ShouldBeSubsetOfMessageGenerator.cs]: Ah yes, sorry for my broken English :stuck_out_tongue: 

I like the third one `{0} should be subset of {1} but {2} are additional` the most.

Comment (JakeGinnivan) [inline src/Shouldly/MessageGenerators/ShouldBeSubsetOfMessageGenerator.cs]: Maybe `{0} should be subset of {1} but {2} are outside subset` as it is quite explicit. 

Then this is getting really nitpicky but a single item should read `is` not `are`. For example

 - `new[] { 1, 2, 5 } should be subset of  [2, 3, 4] but [1, 5] are outside subset`
 - `new[] { 2, 3, 5 } should be subset of  [2, 3, 4] but [1] is outside subset`

**Comment by gabrielweyer** [comment]
Let me know what you think.

**Comment by JakeGinnivan** [comment]
Great! Thanks for persisting.

Something has gone funky with the PR so I rebased and merged manually at https://github.com/shouldly/shouldly/commit/59a303e301bdc89bc9678684262700f6ca0ea701

Suggest doing a fetch then `git reset --hard upstream/master` to clean up your master before doing anything else

**Pull request closed without merging**